src/backend/distributed/commands/multi_copy.c

```diff
@@ -560,7 +561,11 @@ CopyToExistingShards(CopyStmt *copyStatement, QueryCompletionCompat *completionT
 
 		dest->receiveSlot(tupleTableSlot, dest);
 
-		processedRowCount += 1;
+		++processedRowCount;
+
+#if PG_VERSION_NUM >= PG_VERSION_14
+		pgstat_progress_update_param(PROGRESS_COPY_TUPLES_PROCESSED, processedRowCount);
```

Comment:
what about local copy?

Reply:
That is not a concern, we only need to consider COPY on the distributed tables.

Reply:
pg_stat_progress_copy doesn't work well for local execution as we re-start many COPY commands in the same backend where the state is cleared for each internal copy: https://github.com/postgres/postgres/blob/609b0652af00374b89411ea2613fd5bb92bca92c/src/backend/commands/copyfrom.c#L1206-L1217